includes/Data/Patterns.php

```diff
@@ -44,27 +44,7 @@ protected static function get_theme_step_patterns() {
 						'active' => true,
 					),
 				),
-				'homepage-styles' => array(
-					'site-header-left-logo-navigation-inline' => array(
-						'active'  => true,
-						'replace' => true,
-					),
-					'homepage-1'  => array(
-						'active' => true,
-						'shown'  => true,
-					),
-					'homepage-2'  => array(
-						'active' => true,
-						'shown'  => true,
-					),
-					'homepage-3'  => array(
-						'active' => true,
-						'shown'  => true,
-					),
-					'site-footer' => array(
-						'active' => true,
-					),
-				),
+				'homepage-styles' => self::get_patterns_for_homepage_menu_slugs(),
```

Comment:
This needs to be a function reference that we can then call in `get_theme_step_patterns_from_step` else it seems like pattern parsing logic is running twice, once in this function call and once when the actual API request comes in and `get_theme_step_patterns_from_step` runs.

Reply:
It's also running on every API request, even for steps other than homepage-styles